lib/jobs/dell-racadm-tool-job.js

```diff
@@ -0,0 +1,126 @@
+
+        this.nodeId = this.context.target;
+        this.action = options.action;
+        this.cifsInfo = {user: options.server_username, password: options.server_password ,
+            filePath: options.server_filePath};
+    }
+    util.inherits(racadmToolJob, BaseJob);
+
+    /**
+     * @memberOf racadmToolJob
+     */
+    racadmToolJob.prototype._run = function() {
+        var self = this;
+        waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function (node) {
+                assert.ok(node, 'No node for dell racadm set bios');
+                var obmSetting = _.find(node.obmSettings, { service: 'ipmi-obm-service' });
+                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm set bios');
+                return obmSetting.config;
+            })
+            .then(self.revealSecrets)
+            .then(self.lookupHost)
+            .then(function(obmSetting) {
+                if (self.action === 'SetBIOS'){
+                    return racadmTool.setBiosConfig(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);
+                } else if (self.action === 'UpdateFirmware'){
+                    return racadmTool.updateFirmware(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);
+                }else if (self.action === 'GetBIOS'){
+                    return racadmTool.getBiosConfig(obmSetting.host, obmSetting.user,
+                        obmSetting.password,
+                        self.cifsInfo);}else{
+                    throw new Error('The action specified is not supported!.');
+                }
+            })
+            .then(function(results){
+                logger.debug(results.status);
+                self._done();
+            })
+            .catch(function(err) {
+                if (self.action === 'SetBIOS') {
+                    logger.error('Setting BIOS with racadm tool failed!', {error: err});
+                }else if(self.action === 'UpdateFirmware') {
+                    logger.error('Updating Idrac image with racadm tool failed!', {error: err});
+                }else if(self.action === 'GetBIOS') {
+                    logger.error('Getting BIOS with racadm tool failed!', {error: err});
+                }
+                else{
+                    err = new Error('The action specified is not supported!.');
+                }
+                self._done(err);
+            });
+
+    };
+
+    racadmToolJob.prototype.revealSecrets = function revealSecrets (options) {
+        if (options.password) {
+            options.password = encryption.decrypt(options.password);
+        }
+
+        if (options.community) {
+            options.host = encryption.decrypt(options.community);
+        }
+        return options;
+    };
+
+    racadmToolJob.prototype.lookupHost = function lookupHost (options) {
```

Comment:
I have seen the helper functions `lookupHost` and 'revelSecrets` are used in multi jobs, so every job have a copy for  these two function and need to write the same unit-test to improve test coverage. So I recommend to move these two function to a common job utility. I can accept not do in this PR, but adds a follow-up action for this enhancement since multi jobs will be impacted.